lib/elastic_record/doctype.rb

```diff
@@ -18,12 +18,12 @@ class Doctype
 
     def initialize(name, mapping = DEFAULT_MAPPING.deep_dup)
       @name = name
-      @mapping = mapping
+      @mapping = mapping.deep_symbolize_keys
       @analysis = {}
     end
 
     def mapping=(custom_mapping)
-      mapping.deep_merge!(custom_mapping)
+      mapping.deep_merge!(custom_mapping.deep_symbolize_keys)
```

Comment:
Is symbolizing keys necessary?

Reply:
Well, _strictly_ speaking, no. But it's certainly a lot safer...I found that our Places mapping had a mixture of string keys and symbol keys before creating the index. And when additional mapping is added, `deep_merge` treats `"key" => {...}` and `:key => {...}` as different and wont merge them. But when converting to JSON and submitting it as a request body to ES, those keys will be the same and be duplicated.

In < ES 6, this was fine, it would just take the most recent duplicated key and use that value. But in ES 6+, duplicate keys will return an error.